examples/step-79/doc/intro.dox

```diff
@@ -0,0 +1,397 @@
+@f[
+  \min_{\rho,\sigma,\mathbf u, s_1, s_2} \int_{\partial\Omega} \textbf{u} \cdot \textbf{f} - \alpha \int_\Omega \left(\log(s_1) + \log(s_2)\right)
+@f]
+@f[
+  \text{s.t.   } \rho = H(\sigma)
+@f]
+@f[
+  \int_\Omega \rho^p \left(\frac{\mu}{2}\left(\epsilon(\textbf{v}):\epsilon(\textbf{u})) \right) + \lambda \left( \nabla \cdot \textbf{u} \nabla \cdot \textbf{v} \right)  \right) = \int_{\partial \Omega} \textbf{v} \cdot \textbf{f}
+@f]
+@f[
+  \int_\Omega \rho = V
+@f]
+@f[
+  \sigma = s_1
+@f]
+@f[
+  1-\sigma = s_2
+@f]
+
+With these variables in place, we can then follow the usual approach to solving constrained optimization problems: We introduce a Lagrangian in which we combine the objective function and the constraints by multiplying the constraints by Lagrange multipliers. Specifically, we will use the following symbols for the Lagrange multipliers for the various constraints:
+\begin{enumerate}
+    \item $\textbf y_1 $ - a Lagrange multiplier corresponding to the elasticity constraint
+    \item $y_2$ - a Lagrange multiplier corresponding to the convolution filter constraint
+    \item $z_1$ - a Lagrange multiplier corresponding to the lower slack
+    \item $z_2$ - a Lagrange multiplier corresponding to the upper slack
+\end{enumerate}
+With these variables, the Lagrangian function reads as follows:
+
+@f[
+  \mathcal{L} = \int_{\partial\Omega} \textbf{u} \cdot \textbf{f}  - \alpha \int_\Omega \left(\log(s_1) + \log(s_2)\right) -  \left(\int_\Omega \rho^p \left(\frac{\mu}{2}\left(\epsilon(\textbf{y}_1):\epsilon(\textbf{u})) \right) + \lambda \left( \nabla \cdot \textbf{u} \nabla \cdot \textbf{y}_1 \right)  \right)- \int_{\partial \Omega} \textbf{y}_1 \cdot \textbf{f}\right) @f]@f[- y_2 (\rho - H(\sigma)) - z_1 (\sigma-s_1) - z_2 (1 - s_2 -\sigma) 
+@f]
+
+The solution of the optimization problem then needs to satisfy what are known as the Karush-Kuhn-Tucker (KKT) conditions: The derivatives of the Lagrangian with respect to all of its arguments need to be equal to zero, and because we have inequality constraints, we also have "complementarity" conditions. Since we here have an infinite-dimensional problem, these conditions all involve directional derivatives of the Lagrangian with regard to certain test functions -- in other words, all of these conditions have to be stated in weak form as is typically the basis for finite element methods anyway.
+
+The barrier method allows us to initially weaken the ``complementary slackness" as required by the typical KKT conditions. Typically, we would require that $s_i z_i = 0$, but the barrier formulations give KKT conditions where $s_i z_i = \alpha$, where $\alpha$ is our barrier parameter. As part of the barrier method, this parameter must be driven close to 0 to give a good approximation of the original problem.
+
+In the following, let us state all of these conditions where
+$d_{\{\cdot\}}$ is a test function that is naturally paired with the $\{\cdot\}$ function.
```

Comment:
This is because of the large numbers of different test functions I would need - u (my displacement) would pair nicely with v, but to make it consistent across all of my functions, I'm using this notation. It also alludes to the derivatives where they came from